packages/roosterjs-editor-dom/lib/utils/extractClipboardEvent.ts

```diff
@@ -45,25 +50,53 @@ export default function extractClipboardEvent(
         rawHtml: undefined,
     };
 
+    const handlers: {
+        promise: Promise<string>;
+        callback: (value: string) => void;
+    }[] = [];
+
     if (event.clipboardData && event.clipboardData.items) {
         event.preventDefault();
-        let items = event.clipboardData.items;
+
+        // Set rawHtml to null so that caller knows that we have tried
+        result.rawHtml = null;
+        const items = event.clipboardData.items;
+
         for (let i = 0; i < items.length; i++) {
             let item = items[i];
-            if (item.type && item.type.indexOf('text/html') == 0) {
-                item.getAsString(html => {
-                    result.rawHtml = Browser.isEdge ? workaroundForEdge(html) : html;
-                    callback(result);
-                });
-                return;
+
+            switch (item.type) {
+                case 'text/html':
+                    handlers.push({
+                        promise: getAsString(item),
+                        callback: value => {
+                            result.rawHtml = Browser.isEdge ? workaroundForEdge(value) : value;
+                        },
+                    });
+                    break;
+                case 'text/link-preview':
+                    if (options?.allowLinkPreview) {
+                        handlers.push({
+                            promise: getAsString(item),
+                            callback: value => {
+                                try {
+                                    result.linkPreview = JSON.parse(value) as EdgeLinkPreview;
+                                } catch {}

```

Comment:
Do we want to hand this at all when there's a failure? Do we have any type of logging?

Reply:
RoosterJs doesn't support log today. Before we log something by overriding some core API. But it is a good idea to have some unified way to do logging. This can be a separate work item.